raycicmd/dep_graph.go

```diff
@@ -0,0 +1,43 @@
+package raycicmd
+
+type depNode interface {
+	deps() []string
+	mark() // mark if this node should be included
+}
+
+type depGraph interface {
+	depNode(id string) depNode
+}
+
+func markDeps(g depGraph, set map[string]struct{}) {
```

Comment:
maybe add docstring to explain what the function and the `set` argument mean

Reply:
added